lib/bundles/inspec-compliance/api.rb

```diff
@@ -27,15 +27,21 @@ def self.profiles(config)
       # TODO, api should not be dependent on .supported?
       response = Compliance::HTTP.get(url, config['token'], config['insecure'], !config.supported?(:oidc))
       data = response.body
-      if !data.nil?
+      response_code = response.code
+      case response_code
+      when '200'
         profiles = JSON.parse(data)
         # iterate over profiles
         profiles.map do |owner, ps|
           ps.keys.map do |name|
             { org: owner, name: name }
           end
         end.flatten
+      when '401'
+        puts '401 Unauthorized. Please check your token.'
+        []
       else
+        puts response_code, 'An error occured'
```

Comment:
> Only the cli class should do that. therefore we need to return the message as the other methods do it.

👍 On this.  I was going to potentially recommending creating a custom exception that you could raise in the case of an HTTP error.  The exception can contain the message the cli class should print on rescue.  However, I noticed that this code base stays away from exceptions for the most part.